### PR TITLE
Stop surfacing response body as part of errors

### PR DESCRIPTION
**Pull request by jladieu**

### CODE REVIEW

File: lib/services/chatwork.rb [hunk 1]
```diff
@@ -49,8 +49,8 @@ def send_message(config, message)
       req.params['body'] = message
     end
     if res.status < 200 || res.status > 299
-      raise "Could not send a message to room. HTTP Error: #{res.status}. #{res.body}"
+      raise "Could not send a message to room - #{error_response_details(res)}"
     end
-    JSON.parse(res.body)
+    return true
```
Comment (samn) [inline lib/services/chatwork.rb]: `return` not needed?

File: lib/services/slack.rb [hunk 2]
```diff
@@ -15,21 +15,31 @@ class Service::Slack < Service::Base
...
+    end
   end
 
   def receive_issue_impact_change(config, payload)
     message, options = format_issue_impact_change_message(payload)
-    send_message(config, message, options)
-    :no_resource
+    response = send_message(config, message, options)
+    if response.code == '200'
+      :no_resource
```
Comment (samn) [inline lib/services/slack.rb]: why `:no_resource` here, vs the `true` in other integrations?

File: lib/services/chatwork.rb (hunk 1 above)
Comment (jladieu) [inline lib/services/chatwork.rb]: fixed

File: lib/services/slack.rb (hunk 2 above)
Comment (jladieu) [inline lib/services/slack.rb]: fixed, I cut over a bunch of these `:no_resource` to use true instead, and updated the base class documentation to reflect this change (and also explain the historic reason that some of the other service hooks still return hashy values)

File: lib/service/http.rb [hunk 3]
```diff
@@ -125,22 +125,6 @@ def http(options = {})
       end
     end
 
-    # Public: Shortens the given URL with bit.ly.
```
Comment (jladieu) [inline lib/service/http.rb]: Not used, so I removed this.

File: README.md [hunk 4]
```diff
@@ -38,16 +38,16 @@ class Service::Foo < Service::Base
 
   # Receives a config hash containing :identifier => value pairs for each input field
   # and a payload which for impact change events is a hash of data about the issue.
-  # Returns a hash with data in it needed to later accesses any resources created
-  # on the service or :no_resource if there wasn't a resource created.
   #
-  # For unexpected errors, it's recommended to raise here and let the integration
```
Comment (jladieu) [inline README.md]: Made some updates here to explain the expected return value

**Comment by jladieu** [comment]
@samn based on your feedback, I cut over all the `issue_impact_change` implementations to just return `true` on success, and updated the contribution docs to reflect this change of recommended behavior; it should be noted the existing harness will keep functioning as it simply checks for a truthy value to determine success

File: README.md (hunk 4 above)
Comment (samn) [inline README.md]: nice!

File: lib/service/http.rb (hunk 3 above)
Comment (samn) [inline lib/service/http.rb]: wonderful

**Comment by samn** [comment]
approved!

**Pull request merged**